Scripts: Added Gullible's Travels

The-Halcyon-Days opened the pull request.

whasf commented:
Don't forget to mark the quest complete in /scripts/globals/quests.lua

whasf commented on scripts/zones/Kazham/npcs/Magriffon.lua:
```diff
@@ -35,9 +59,14 @@ end;
+    if(csid == 0x0090 and option == 1)  then                 -- Gullible's Travels: First CS
+        player:addQuest(OUTLANDS, GULLIBLES_TRAVELS);
+    elseif(csid == 0x0092) then                              -- Gullible's Travels: Final CS
+        player:confirmTrade();
```
You didn't use confirmItem(), why are you using confirmTrade()?

whasf commented on scripts/zones/Kazham/npcs/Magriffon.lua:
```diff
@@ -1,25 +1,49 @@
-end; 
+    if(player:getQuestStatus(OUTLANDS, GULLIBLES_TRAVELS) == QUEST_ACCEPTED) then
+        if(trade:getGil() >= player:getVar("MAGRIFFON_GIL_REQUEST")) then
+            player:startEvent(0x0092);
```
Maybe tradeComplete() here? I think we've had problems in the past trying to finish a trade across functions/events

teschnei replied:
I told him to use confirmTrade because tradeComplete will delete the gil
you traded (instead of just the amount you owed)

On Tue, Jul 14, 2015 at 5:36 PM, whasf <notifications@github.com> wrote:

> In scripts/zones/Kazham/npcs/Magriffon.lua
> <https://github.com/DarkstarProject/darkstar/pull/1718#discussion_r34632616>
> :
>
> >  -----------------------------------
> >  -- onTrade Action
> >  -----------------------------------
> >
> >  function onTrade(player,npc,trade)
> > -end;
> > +    if(player:getQuestStatus(OUTLANDS, GULLIBLES_TRAVELS) == QUEST_ACCEPTED) then
> > +        if(trade:getGil() >= player:getVar("MAGRIFFON_GIL_REQUEST")) then
> > +            player:startEvent(0x0092);
>
> Maybe tradeComplete() here? I think we've had problems in the past trying
> to finish a trade across functions/events
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DarkstarProject/darkstar/pull/1718/files#r34632616>.
>

The-Halcyon-Days commented:
Alright. I added it to the quest.lua. I didn't see a legend for how to mark a quest complete so I just added a + beside it like it seemed some of the other quests I saw did. Is the confirmTrade() use ok afterall?

The-Halcyon-Days commented:
Fixed the issue I had with the quest.lua encoding. Everything should be ok now.
